src/libcore/iter/range.rs

```diff
@@ -89,6 +89,7 @@ macro_rules! step_impl_unsigned {
             }
 
             #[inline]
+            #[allow(unreachable_patterns)]
```

Comment:
Why not just remove the `Err` branch?

Reply:
Because it's only unreachable for certain `$t`s (like `u128`), not all of them (like `u8`).